book/src/programmability/witness-pattern.md

```diff
@@ -1,277 +1,117 @@
+    /// A struct that requires a witness to be created.
+    public struct Instance<T> { t: T }
 
-/// Something is missing here! Anyone can create a `Coin<T>` with any `T`.
-public fun new_coin<T>(value: u64): Coin<T> { Coin { value } }
-```
-
-While the `new_container` function requires the `T` value to be passed, the `new_coin` function is
-missing this part. And this implementation of the `new_coin` allows anyone to create a `Coin<T>`
-with any `T`. And, naturally, this is not the behaviour we want.
-
-Now, how can we fix this? We want to ensure that only a specific module can create a `Coin<T>` with
-its `T`. Otherwise, the type signature of `Coin<T>` makes little sense. And similar to the
-`Container<T>`, we need a way to present some `T` to the `new_coin` function. And given that this
-`T` is not stored in the struct and not stored to be used later (important security property), we
-would want to use a type that has the `drop` ability - a Witness.
-
-An updated signature of the `new_coin` would then look like this:
-
-```move
-/// Requires a `T` with the `drop` ability as a proof of ownership of `T`.
-/// However, the value is not used and discarded right away.
-public fun new_coin<T: drop>(_witness: T, value: u64): Coin<T> {
-    Coin { value }
+    /// Create a new instance of `Instance<T>` with the provided T.
+    public fun new<T>(witness: T): Instance<T> {
+        Instance { t: witness }
+    }
 }
 ```
 
-And unlike the `Container<T>`, where the `T` can be any value, including an
-[object](./../storage/key-ability.md) (hence, it can be supplied _dynamically_), the `T` in the
-`new_coin` function has to be created and consumed _statically_ - by implementing a function in the
-module that owns the `T`. So, if the type `MyCoin` is defined in the `my_coin` module, only this
-module can create a new `Coin` with `T` being `MyCoin`.
+The only way to construct an `Instance<T>` is to call the `new` function with an instance of the
+type `T`. This is a basic example of the witness pattern in Move. A module providing a witness often
+has a matching implementation, like the module `book::witness_source` below:
 
 ```move
-module book::my_coin {
-    use book::simple_coin::{Self, Coin};
+module book::witness_source {
+    use book::witness::{Self, Instance};
 
-    /// The Witness type for the Coin.
-    public struct MyCoin has drop {}
+    /// A struct used as a witness.
+    public struct W {}
 
-    /// The implementation of the `new_coin` function. Witness is not storable,
-    /// thus the only way to create a `Coin<MyCoin>` is to call this function in
-    /// this module.
-    public fun new_mycoin(value: u64): Coin<MyCoin> {
-        simple_coin::new_coin(MyCoin {}, value)
+    /// Create a new instance of `Instance<W>`.
+    public fun new_instance(): Instance<W> {
+        witness::new(W {})
     }
 }
 ```
 
-## "Abstract Class"
+The instance of the struct `W` is passed into the `new` function to create an `Instance<W>`, thereby
```

Comment:
Is it worth reinforcing here that if you are given an `Instance<W>` you know that it came from the module that defined `W` as long as there isn't a public function in that module that constructs values of type `W`?